sql/2020/01_CSS/all_functions.sql

```diff
@@ -0,0 +1,54 @@
+#standardSQL
+CREATE TEMPORARY FUNCTION getProperties(css STRING) RETURNS ARRAY<STRING> LANGUAGE js AS '''
+try {
+	function compute() {
+		let ret = {};
+
+		walkDeclarations(ast, ({property, value}) => {
+			if (value.length > 1000 || !value.includes("(") || !value.includes(")")) {
+				return;
+			}
+
+			for (let {name} of extractFunctionCalls(value)) {
+				incrementByKey(ret, name);
+			}
+		});
+
+		return sortObject(ret);
+
+	}
+
+	let ast = JSON.parse(css);
+	let props = compute(ast);
+	return Object.entries(props).flatMap(([prop, freq]) => {
+		return Array(freq).fill(prop);
+	});
+}
+catch (e) {
+	return [];
+}
+'''
+OPTIONS (library="gs://httparchive/lib/css-utils.js");
+
+SELECT
+  *
+FROM (
+  SELECT
+    client,
+    prop,
+    COUNT(DISTINCT page) AS pages,
```

Comment:
Are you interested in this as % of pages as well, or is the absolute number useful enough for context? Not super important to calculate in the query as opposed to the sheet though.

Reply:
I've been adding computed columns to the sheets with the total # of pages and pct_pages. Doing it in the query would be preferable, if it doesn't increase complexity too much.